### DISCUSSION TITLE
Criticism: Does VitePress have any benefit in cases where heavy theme customization is used?

### DISCUSSION THREAD

**Discussion by StarLederer** [vuejs/vitepress#1986] [General]
I've been trying VitePress out as an alternative to my own documentation frontend that I built but so far I got the impression that VitePress is only helpful if the default theme is used. I would like to share this view and see if there is anything I missed that mislead me into thinking that or if this can help the VitePress theme further improve the project.

VitePress works great for cookie-cutter documentation which is what you need in most cases, but if heavy customization is required for marketing and the whole theme is made from scratch does VitePress actually do anything meaningful? I know it comes with nice file based routing, but if I am not using any of the default theme components does the whole benefit come down to an opinionated config file? Wouldn't it in that case be better for me to just build a normal Vite project with [vite-plugin-md](https://github.com/antfu/vite-plugin-md) and [unplugin-vue-router](https://github.com/posva/unplugin-vue-router), and reserve the freedom to go with other frameworks while at it?

Please let me know if I misunderstood something or if I'm missing a part of the picture. I'm really hoping to change my mind about this

**Comment by kiaking** [comment]
VitePress is a SSG (Static Site Generation) framework. So, you get server side rendered HTML on build time. If you build website with plain Vite, then you get SPA.

On the other hand, it does not support SSR like Nuxt does (VitePress don't have a server to serve on production).

Me personally, using VitePress for SSG website. Like this one: https://xlimit.globalbrains.com

It's not using VitePress theme at all. The reason why I use VitePtess, is to get SSG feature. I'm not even using any Markdown for this site.

If you don't need SSG, and you just want to create SPA, then I would recommend using plain Vite as you mentioned. But if you want SSG, I think VitePress is a quite nice choice 👍

**Reply by StarLederer** [reply]
Oh, right... I totally missed the fact that it targets SSG. I guess I do see how it is simpler than Nuxt for this case. I'm glad to know that SSG is the core feature because that makes sense now but I am also relieved to know that it is totally fine to choose something else if my requirements diverge.

Thank you for your answer
